src/middlewared/middlewared/etc_files/local/collectd.conf

```diff
@@ -0,0 +1,170 @@
+<%
+	if not middleware.call_sync('systemdataset.update_collectd_dataset'):
+		# Let's exit this if collectd related disk operations fail
+		middleware.logger.error('Collectd configuration file could not be generated')
+		exit(1)
```

Comment:
Did you test exit? I didnt find it in mako reference docs.

Reply:
Not yet, I am building an iso. Will test this there